### ISSUE TITLE
Wrong data type using MAKE_ARRAY

### ISSUE THREAD

**Issue by sflinois** [gnudatalanguage/gdl#310]
MAKE_ARRAY in GDL seems to generate an array based on the value and not the type we are using !

```
GDL> ARR = MAKE_ARRAY(10, 1, /INT, VALUE = 5)
GDL> help, ARR
ARR             INT       = Array[10]
GDL> ARR = MAKE_ARRAY(10, 1, /INT, VALUE = 5.0)
GDL> help, ARR
ARR             FLOAT     = Array[10]
GDL> ARR = MAKE_ARRAY(10, 1, /FLOAT, VALUE = 5)
GDL> help, ARR
ARR             INT       = Array[10]
```

IDL handle those cases normaly 

```
IDL> ARR = MAKE_ARRAY(10, 1, /INT, VALUE = 5)
IDL> help, ARR
ARR             INT       = Array[10]
IDL> ARR = MAKE_ARRAY(10, 1, /INT, VALUE = 5.0)
IDL> help, ARR
ARR             INT       = Array[10]
IDL> ARR = MAKE_ARRAY(10, 1, /FLOAT, VALUE = 5)
IDL> help, ARR
ARR             FLOAT     = Array[10]
```

**Comment by alaingdl** [comment]
"bien vu" ! 
Again it's indicated that the tests are far from exhaustive on basic operations :(

we do have "give_list_numeric.pro" to generate types and use that in loops
(already used in TEST_FIX, TEST_TOTAL (TBC), TEST_ROUNDING ...)

**Comment by rexso** [comment]
Coincidentally, I noticed this bug as well yesterday while working on implementing the **START** and **INCREMENT** keywords, so I have fixed this issue. Will submit a patch later today after writing some tests.

**Comment by alaingdl** [comment]
I already wrote some "general" tests, to be public in few hours

**Comment by rexso** [comment]
This problem should now be fixed with my latest PR.

@alaingdl feel free to merge your tests with the basic ones I wrote. I do not have much IDL experience, so more experienced IDL users familiar with MAKE_ARRAY could probably improve my tests quite a lot.

**Comment by alaingdl** [comment]
@rexso I am trying to merge ! some of the tests are equivalent, it is not really important.

I just would like to indicate : 
1/ please use ERRORS_ADD instead ADD_ERROR, I hope to change all ADD_ERROR for ERRORS_ADD
(consistancy : we have also ERRORS_CUMUL & ERRORS_RESET)

2/ syntax "if NOT something then" should be avoid, "if ~something then" is way better
(bitwise nature of the NOT operator) Discussion here [Logical operators](http://www.harrisgeospatial.com/docs/Logical_Operators.html) and here [Bitwise operators](http://www.harrisgeospatial.com/docs/Bitwise_Operators.html#expressions_2976722315_1057479)

**Comment by rexso** [comment]
@alaingdl thank you for the heads up regarding the operators. I was not aware that **NOT** is *bitwise* whereas **~** is *logical*. Quite the gotcha when primarily programming in C and related languages. I will keep this in mind.

**Comment by rexso** [comment]
Fixed in #316 as verified by tests written by @alaingdl
